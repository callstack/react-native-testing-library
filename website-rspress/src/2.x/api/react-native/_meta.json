[
  {
    "type": "file",
    "name": "index",
    "label": "Table of Contents"
  },
  {
    "type": "dir",
    "name": "classes",
    "label": "Classes"
  },
  {
    "type": "dir",
    "name": "interfaces",
    "label": "Interfaces"
  },
  {
    "type": "dir",
    "name": "types",
    "label": "Types"
  }
]
