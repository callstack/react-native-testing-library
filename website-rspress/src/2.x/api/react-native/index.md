# Re.Pack React Native APIs

## Table of contents

### Classes

- [Chunk](./classes/Chunk.md)
- [ChunkManager](./classes/ChunkManager.md)

### Interfaces

- [ChunkConfig](./interfaces/ChunkConfig.md)
- [ChunkManagerConfig](./interfaces/ChunkManagerConfig.md)
- [RemoteChunkLocation](./interfaces/RemoteChunkLocation.md)
- [StorageApi](./interfaces/StorageApi.md)

### Type aliases

- [RemoteChunkResolver](./types/RemoteChunkResolver.md)
