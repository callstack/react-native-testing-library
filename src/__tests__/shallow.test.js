// @flow
import React from 'react';
import { View, Text } from 'react-native';
import { shallow, render } from '..';

const TextPress = () => (
  <View>
    <Text testID="text-button">Press me</Text>
  </View>
);

test('shallow rendering React elements', () => {
  const { output } = shallow(<TextPress />);

  expect(output).toMatchSnapshot();
});

test('shallow rendering React Test Instance', () => {
  const { getByTestId } = render(<TextPress />);
  const { output } = shallow(getByTestId('text-button'));

  expect(output).toMatchSnapshot();
});
