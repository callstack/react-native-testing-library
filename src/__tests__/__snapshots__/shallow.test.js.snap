// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow rendering React Test Instance 1`] = `
<TouchableText
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  forwardedRef={null}
  testID="text-button"
>
  Press me
</TouchableText>
`;

exports[`shallow rendering React elements 1`] = `
<Component>
  <Component
    testID="text-button"
  >
    Press me
  </Component>
</Component>
`;
